src/evo/deterministicmns.cpp

```diff
@@ -809,122 +810,122 @@ bool CDeterministicMNManager::BuildNewListFromBlock(const CBlock& block, gsl::no
                     __func__, tx.GetHash().ToString(), nHeight, proTx.ToString());
             }
         } else if (tx.nType == TRANSACTION_PROVIDER_UPDATE_SERVICE) {
-            CProUpServTx proTx;
-            if (!GetTxPayload(tx, proTx)) {
+            const auto opt_proTx = GetTxPayload<CProUpServTx>(tx);
+            if (!opt_proTx) {
                 return state.Invalid(BlockValidationResult::BLOCK_CONSENSUS, "bad-protx-payload");
             }
 
-            if (proTx.nType == MnType::Evo && !DeploymentActiveAfter(pindexPrev, Params().GetConsensus(), Consensus::DEPLOYMENT_V19)) {
+            if (opt_proTx->nType == MnType::Evo && !DeploymentActiveAfter(pindexPrev, Params().GetConsensus(), Consensus::DEPLOYMENT_V19)) {
```

Comment:
nit: for reduce size of diff can introduce here `const auto& proTx = *opt_proTx;`

Reply:
just for reference:
https://godbolt.org/z/5GqYbMWz6

The gcc (version 8, 13) without any optimization (-O0) acts literally as requested and there's tiny overhead: it does 2 extra memory-register copy for this operation.

The gcc (version 8, 13) with minimal optimization option (-O1) generate completely identical code with and without any overhead; assembler code identical.

Clang 17 does same with -O0 (2 extra copies between register and memory) and with -O1 (identical code)